pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1539,6 +1539,34 @@ class Timedelta(_Timedelta):
         div = other // self
         return div, other - div * self
 
+    # GH45129
+    def total_seconds(self, *, ns_precision=False) -> float:
+        """
+        Total seconds in the duration with default us precision
+        (for compatibility with `datetime.timedelta`).
+
+        Parameters
+        ----------
+        ns_precision : bool, default False
+            Return the duration with ns precision.
+
+        Examples
+        --------
+        >>> td = pd.Timedelta(days=6, minutes=50, seconds=3,
+        ...                   milliseconds=10, microseconds=10, nanoseconds=12)
+        >>> td
+        Timedelta('6 days 00:50:03.010010012')
+
+        >>> td.total_seconds()
+        521403.01001
+
+        >>> td.total_seconds(ns_precision=True)
+        521403.010010012
+        """
+        if ns_precision:
```

Comment:
umm what is this?

Reply:
I could not find a way to prevent the problems that the higher precision causes in timezone-related-info (see link from @mroeschke) and because of that I am keeping the `super` implementation the default. However, having a `total_seconds` method unable to output nanosecond precision is not consistent. My solution for this was to override the method with this version that has a documented `kwarg` such that anybody can then know about it (possibly after a bad surprise) without having to check source code.

Reply:
adding a keyword like this is not going to fly. if you want to try to fix that issue then happy to incorporate here (or in a pre-cursor). that looks like the test case needs to be updated.

Reply:
Agreed. The signatures need to be kept compatible with the `datetime.timedelta` methods as `Timedelta` is a subclass

Reply:
Ok. Will retry at a later time.